derive_builder_core/src/macro_options/darling_opts.rs

```diff
@@ -95,6 +95,25 @@ pub struct BuildFn {
     /// * If `validate` is specified, then this type must provide a conversion from the specified
     ///   function's error type.
     error: Option<Path>,
+    /// Whether to generate `ValidationError(String)` as a variant of the build error type.
+    ///
+    /// Setting this to `false` will prevent `derive_builder` from using the `validate` function
+    /// but this also means it does not generate any usage of the `alloc` crate (useful when
+    /// disabling the `alloc` feature in `no_std`).
+    validation_error: bool,
+}
+
+impl BuildFn {
+    fn validation_needs_error(self) -> darling::Result<Self> {
```

Comment:
We should make sure someone doesn't use both `validation_error` and `error` in this method too. (And use `darling::Error::accumulator` to make sure we get all the errors gathered up at once before returning)

Reply:
I moved to the accumulator in case future stuff wants to use this method. `error` and `validation_error` is now impossible via the parse since I'm using an enum for it (from your other comment).